src/GraphServiceClient.php

```diff
@@ -22,11 +25,35 @@
 class GraphServiceClient extends Generated\BaseGraphClient
 {
     /**
+     * @var RequestAdapter|null
+     */
+    private static ?RequestAdapter $customRequestAdapter = null;
+
+    /**
+     * @param TokenRequestContext $tokenRequestContext
+     * @param array $scopes
+     */
+    public function __construct(TokenRequestContext $tokenRequestContext, array $scopes = [])
+    {
+        if (self::$customRequestAdapter) {
+            parent::__construct(self::$customRequestAdapter);
+        } else {
+            parent::__construct(new GraphRequestAdapter(
+                new GraphPhpLeagueAuthenticationProvider($tokenRequestContext, $scopes)
+            ));
+        }
+    }
+
+    /**
+     * Get an instance of GraphServiceClient that uses $requestAdapter
      * @param RequestAdapter $requestAdapter
+     * @return GraphServiceClient
      */
-    public function __construct(RequestAdapter $requestAdapter)
+    public static function createWithRequestAdapter(RequestAdapter $requestAdapter): GraphServiceClient
```

Comment:
What scenario would this cover? I don't think we need this and a constructor.

Reply:
Okay, I'm now going to read the issue that is driving this...... to answer my questions.....

Reply:
I now see. I don't think we want to set the TokenRequestContext after we've created the GraphServiceClient. Once we have a GraphServiceClient instance, we shouldn't be making changes to the client.

Reply:
We're only supporting alternatives to creating a new instance of the client. Updating the instance isn't possible. Using `GraphServiceClient::createWithRequestAdapter()` or `new GraphServiceClient()` creates separate instances.

No setters are available on the relevant objects during instantiation i.e. GraphServiceClient, RequestAdapter, Authentication provider and token request context objects don't support updating an existing instance. They can only be initialised once via constructors. Unless the developer provides their own implementation of the interfaces & adds setters.